lib/src/value/calculation.dart

```diff
@@ -120,6 +125,186 @@ class SassCalculation extends Value {
     return SassCalculation._("max", args);
   }
 
+  /// Creates a `hypot()` calculation with the given [arguments].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation]. It must be passed at least one argument.
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  static Value hypot(Iterable<Object> arguments) {
+    var args = _simplifyArguments(arguments);
+    if (args.isEmpty) {
+      throw ArgumentError("hypot() must have at least one argument.");
+    }
+    _verifyCompatibleNumbers(args);
+
+    var subtotal = 0.0;
+    var first = args.first;
+    if (first is! SassNumber || first.hasUnit('%')) {
+      return SassCalculation._("hypot", args);
+    }
+    for (var i = 0; i < args.length; i++) {
+      var number = args.elementAt(i);
+      if (number is! SassNumber || (!number.hasCompatibleUnits(first))) {
```

Comment:
Unnecessary parentheses:

```suggestion
      if (number is! SassNumber || !number.hasCompatibleUnits(first)) {
```